packages/react-dom/src/events/ReactDOMEventListener.js

```diff
@@ -125,70 +136,66 @@ export function isEnabled() {
   return _enabled;
 }
 
-/**
- * Traps top-level events by using event bubbling.
- *
- * @param {number} topLevelType Number from `TopLevelEventTypes`.
- * @param {object} element Element on which to attach listener.
- * @return {?object} An object with a remove function which will forcefully
- *                  remove the listener.
- * @internal
- */
 export function trapBubbledEvent(
   topLevelType: DOMTopLevelEventType,
-  element: Document | Element,
-) {
-  if (!element) {
-    return null;
-  }
+  element: Document | Element | Node,
+  isLegacy: boolean,
+): void {
   const dispatch = isInteractiveTopLevelEventType(topLevelType)
     ? dispatchInteractiveEvent
     : dispatchEvent;
+  const rawEventName = getRawEventName(topLevelType);
 
-  addEventBubbleListener(
-    element,
-    getRawEventName(topLevelType),
+  if (isLegacy) {
     // Check if interactive and wrap in interactiveUpdates
-    dispatch.bind(null, topLevelType),
-  );
+    const listener = dispatch.bind(null, topLevelType, null);
+    // We don't listen for passive/non-passive
+    addEventBubbleListener(element, rawEventName, listener, null);
+  } else {
+    // Check if interactive and wrap in interactiveUpdates
+    const passiveListener = dispatch.bind(null, topLevelType, true);
+    const activeListener = dispatch.bind(null, topLevelType, false);
+    // We listen to the same event for both passive/non-passive
+    addEventBubbleListener(element, rawEventName, passiveListener, true);
+    addEventBubbleListener(element, rawEventName, activeListener, false);
```

Comment:
If the browser doesn't support passive/non passive events, we would potentially adding two _different_ (since we used `.bind`) event listeners here so they might double fire. This is of course pure speculation but it might need a feature test here as well (and for the capture case).

Reply:
This is intentional. We want to double listen so the new event APIs in the future can control heuristics given both. They'd likely guard one path given an API. For example `onTouchMovePassive` vs `onTouchMoveActive`.

Reply:
Yeah you're right the listeners will only call the respective callbacks so we need them both to fire. My bad. :-)